### PR TITLE
ignore unset pTdnf->pArgs in TDNFRefresh()

### PR DESCRIPTION
**Pull request by oliverkurth**
As requested by @shreyasbe , this is to fix pmd.

### CODE REVIEW

File: client/init.c [hunk 1]
```diff
@@ -291,6 +291,13 @@ uint32_t
 TDNFRefresh(
     PTDNF pTdnf)
 {
-    return TDNFRefreshSack(pTdnf, pTdnf->pSack, pTdnf->pArgs->nRefresh);
+    if (pTdnf->pArgs)
```
Comment (sshedi) [inline client/init.c]: Please validate pTdnf & pTdnf->pSack.

File: client/init.c [hunk 2]
```diff
@@ -291,6 +291,13 @@ uint32_t
...
+    {
+        return TDNFRefreshSack(pTdnf, pTdnf->pSack, pTdnf->pArgs->nRefresh);
+    }
+    else
```
Comment (sshedi) [inline client/init.c]: else is redundant here.

**Review by sshedi — CHANGES_REQUESTED** [review]

**Comment by oliverkurth** [comment]
So looks like I broke the API with PR206 - it caused the TDNFRefresh() call to be made in the client (!), and this of course failed. I now moved the calls to `TDNFRefresh()` to client/api.c where they belong. This makes the calls to be made by the server. Similarly, the call to `TDNFAddCmdLinePackages()` had to be moved.

I tested this various invocations of `pmd-cli` when built with these changes and they looked okay. The `pmd-cli pkg check` command segfaults, but I believe this was broken before. @tapakund , please rebuild pmd with these changes and test thoroughly.

**Comment by jaankit** [comment]
> So looks like I broke the API with PR206 - it caused the TDNFRefresh() call to be made in the client (!), and this of course failed. I now moved the calls to `TDNFRefresh()` to client/api.c where they belong. This makes the calls to be made by the server. Similarly, the call to `TDNFAddCmdLinePackages()` had to be moved.
> 
> I tested this various invocations of `pmd-cli` when built with these changes and they looked okay. The `pmd-cli pkg check` command segfaults, but I believe this was broken before. @tapakund , please rebuild pmd with these changes and test thoroughly.

"pmd-cli pkg check" is failing because corresponding API is not implemented(pkg_invoke_check()) in pmd when it gets called from tdnf but "check" is added in cmdlist at here: https://github.com/vmware/pmd/blob/1ffd119b8831829b30714e3efd6ad48b86c93c9f/tools/cli/pkgmgmt/pkg_main.c#L29 . Thus, it is crashing. This need to be fixed in pmd. This issue was present in old version of pmd also.

Tested some of the command with this change. Works fine. Thorough testing is in progress.

File: client/init.c [hunk 3]
```diff
@@ -291,6 +291,10 @@ uint32_t
 TDNFRefresh(
     PTDNF pTdnf)
 {
+    if (!pTdnf->pArgs)
```
Comment (tapakund) [inline client/init.c]: Can we modify this function to:
uint32_t
TDNFRefresh(
    PTDNF pTdnf)
{
    uint32_t dwError = 0;
    if(!pTdnf || !pTdnf->pSack || !pTdnf->pArgs)
    {
        dwError = ERROR_TDNF_INVALID_PARAMETER;
        return dwError;
    }
    return TDNFRefreshSack(pTdnf, pTdnf->pSack, pTdnf->pArgs->nRefresh);
}

**Review by tapakund — CHANGES_REQUESTED** [review]
All other changes looks fine.

File: client/init.c (hunk 3 above)
Comment (oliverkurth) [reply client/init.c]: Done.

**Pull request merged**